mdakit_membcurv/tests/test_mdakit_membcurv.py

```diff
@@ -3,10 +3,290 @@
+                                                2.69411991e-03, -1.30866347e-02, -2.52293095e-02,
+                                                -7.22431515e-03, -6.73201764e-03, -2.51286123e-01,
+                                                -3.22793339e-01],
+                                               [-2.24613530e-02, 7.82002286e-04, -3.98911117e-04,
+                                                5.29635968e-04, -4.19699155e-03, -1.30479296e-02,
+                                                -2.38928651e-04, 4.24992903e-02, 1.70628313e-02,
+                                                -4.85138439e-02],
+                                               [-1.94220106e-02, -2.13212988e-04, 1.05435416e-03,
+                                                1.30710748e-03, -3.49076390e-03, -1.28226749e-02,
+                                                -1.41168166e-02, -1.08870941e-02, -1.21323346e+00,
+                                                -1.55778010e-01]])}
+}
+
+
+@pytest.fixture()
+def universe():
+    u = mda.Universe(GRO_MEMBRANE_PROTEIN, XTC_MEMBRANE_PROTEIN)
+    return u
+
+
+@pytest.fixture()
+def ref_beads():
+    u = mda.Universe(GRO_PO4)
+    ref_beads = u.atoms[0:455]
+    return ref_beads
+
+
+@pytest.fixture()
+def Z_cloud(ref_beads):
+    n_cells, max_width = 10, 19
+
+    # set grid_1. z coordinates for [i,j] cells.
+    # set grid_2. Count number of beads populating each [i,j] cell.
+    # set z_ref. Final values of average z per [i,j] cell.
+    grid_1, grid_2, z_ref = [np.zeros([n_cells, n_cells]) for i in range(3)]
+
+    # factor used to map (x,y) to [i,j]
+    factor = np.float32(n_cells / max_width)
+
+    # iterate over the bead of reference
+    for atom in ref_beads:
+
+        # extract positions in nm.
+        x, y, z = atom.position / 10
+
+        # map (x,y) to [i,j]
+        cell_l = int(abs(x) * factor)
+        cell_m = int(abs(y) * factor)
+
+        # sum value of z coordinate
+        grid_1[cell_l, cell_m] += z
+        grid_2[cell_l, cell_m] += 1
+
+    # To calculate average, iterate over each cell in the grid:
+    for i, j in it.product(range(n_cells), range(n_cells)):
+        # if the element [i,j] is not empty
+        if grid_2[i, j] > 0:
+            # then calculate the average of z:
+            # grid_1 has the sum of all the z coordinates.
+            # grid_2 counted how many beads were in that grid.
+            z_ref[i, j] += grid_1[i, j] / grid_2[i, j]
+        else:
+            # if there are not beads in that cell, store a nan.
+            z_ref[i, j] = np.nan
+
+    return z_ref
+
+
+@pytest.fixture()
+def mdtraj_po4():
+    # trajectory PO4 beads only imported using mdtraj, gone after refactoring
+    mdtraj = md.load(XTC_PO4, top=GRO_PO4)
+    return mdtraj
+
+
+@pytest.fixture(scope="session")
+def output(tmpdir_factory):
+    tmp = tmpdir_factory.mktemp("output")
+    return tmp
```

Comment:
Instead of this, you could use `tmpdir` directly in your function.

Reply:
Thanks, I didn't realize this. Changed now. It's gone now, but a good exercise.